pybamm/expression_tree/concatenations.py

```diff
@@ -144,25 +144,22 @@ def process_node_for_concatenate(self, node, mesh):
             the sub-expression to process (node.is_constant() is true)
 
         """
-
-        # node must be constant
-        value = node.evaluate()
-
-        # correct size of vector should be number of points in the domains
-        subvector_size = sum([mesh[dom].npts for dom in node.domain])
-
-        # check if its a scalar, if so convert to vector
-        if isinstance(value, numbers.Number):
-            value = np.full(subvector_size, value)
-
-        # check it is the right size
-        if value.size != subvector_size:
-            raise ValueError(
-                "Error: expression evaluated to a vector of incorrect length"
-            )
-
-        # convert to a Vector node
-        return pybamm.Vector(value, domain=node.domain)
+        try:
+            node_size = node.size
+        except AttributeError:
+            node_size = 0
+
+        if node_size > 1:
+            if node_size == sum([mesh[dom].npts for dom in node.domain]):
```

Comment:
I guess this follows on from how we name the components of the mesh but should we be passing the mesh to this function if it doesn't belong to a specific SpatialMethod Class?